src/core/components/ChatViewContainer.tsx

```diff
@@ -0,0 +1,58 @@
+import { ChatView } from './ChatView';
+import { ModelConfig, SupportedModels } from '../types/models';
+import type { ChatCompletionMessageParam } from 'openai/resources';
+import { useMessageHistory } from '../hooks/useMessageHistory';
+import { ActiveConversation } from '../context/types';
+
+interface ChatViewContainerProps {
+  onMenu: () => void;
+  onPanelOpen: () => void;
+  isPanelOpen: boolean;
+  supportsUpload: boolean;
+  showModelSelector: boolean;
+  startNewChat: () => void;
+  conversationID: string;
+  modelConfig: ModelConfig;
+  activeConversation: ActiveConversation;
+  handleChatCompletion: (value: ChatCompletionMessageParam[]) => void;
+  handleCancel: () => void;
+  handleModelChange: (modelName: SupportedModels) => void;
+}
+export const ChatViewContainer = ({
+  onMenu,
+  onPanelOpen,
+  isPanelOpen,
+  supportsUpload,
+  handleModelChange,
+  handleCancel,
+  handleChatCompletion,
+  activeConversation,
+  startNewChat,
+  modelConfig,
+  conversationID,
+}: ChatViewContainerProps) => {
+  const { messages } = useMessageHistory(
+    activeConversation.messageHistoryStore,
+  );
+
+  return (
+    <ChatView
+      handleModelChange={handleModelChange}
+      thinkingStore={activeConversation.thinkingStore}
+      isRequesting={activeConversation.isRequesting}
+      errorStore={activeConversation.errorStore}
+      messageStore={activeConversation.messageStore}
```

Comment:
Follow on PR's to come to subscribe to these stores and pass down more primitive props